Monika After Story/game/zz_hangman.rpy

```diff
@@ -724,10 +675,110 @@ label mas_hangman_game_loop:
 
         "No.":
             pass
+    jump mas_hangman_game_end
 
-            #FALL THROUGH
+label mas_hangman_game_win(guesses=0, chances=0, last_chance_guesses=0):
+    if is_window_sayori_visible:
+        show hm_s_win_6 as window_sayori at hangman_sayori_h
+
+    if player_word:
+        $ the_word = "your name"
+    else:
+        $ the_word = "the word"
+        
+    if chances == 6:
+        m 1sud "Wow, [player]..."
+        if not player_word:
+            m 1hua "You didn't miss a single letter!"
+            m 3eub "You must have a really good vocabulary, ahaha."
+            m 1tsa "...Or maybe you just lucked out, hmm?"
+        else:
+            m 1hkb "You just guessed your own name flawlessly, ahaha."
+    elif chances == 1 and last_chance_guesses / (guesses - (6 - chances)) > 0.5:
+        # More than 50% of the guesses were made on last chance.
+        m 1sud "You just made it on the last chance!"
+        m 1hub "Wow, [player], you must be {i}really{/i} lucky, ahaha~"
+
+        # Make her mention previous episode of giving up on last chance just once.
+        if (
+            mas_seenLabels(["mas_hangman_give_up_last_chance"]) and
+            not mas_seenLabels(["mas_hangman_give_up_last_chance_ref"])
+        ):
+            call mas_hangman_give_up_last_chance_ref
+    else:
+        m 1hua "Wow, you guessed [the_word] correctly!"
+        m "Good job, [player]!"
+
+    if not persistent.ever_won['hangman']:
+        $ persistent.ever_won['hangman'] = True
+        $ mas_gainAffection(2)
+    return
 
-    # RETURN AT END
+label mas_hangman_game_loss(guesses=0, chances=0, last_chance_guesses=0):
+    if guesses > 0:
+        if player_word:
+             m 1eka "[player]..."
+             m "You couldn't guess your own name?"
+        elif last_chance_guesses / (guesses - (6 - chances)) > 0.5:
+             if not (persistent._mas_pm_cares_about_dokis or persistent._mas_sensitive_mode):
+                 m 1tsa "You were {i}hanging{/i} in there really well{w=0.5}{nw}, "
+                 extend 1hub "[player], ahaha."
+             else:
+                 m 1eua "You were doing really great, [player]!"
+        m 1hua "Better luck next time~"
+    else:
+        m 1ekc "[player]..."
+        m "You didn't guess even a single letter..."
+        m 1eka "I really hope you'll do better next time."
+        if mas_isMoniAff(higher=True):
+            m 1ekbsa "Would you try a little harder for me?~"
```

Comment:
m 1hksdlb "Did I pick a word that was too difficult for you, [mas_get_player_nickname()]?~"

Frustrating to hear that you're not trying hard enough if you are. I'm not against Monika saying it, per se, but I just think something in my suggestion is more playful and lighthearted for Aff range. At least it's not the like times below where [player] gives up, so Monika would probably say something more positive here I feel.